src/codecs/webp/decoder.rs

```diff
@@ -169,16 +168,16 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for WebPDecoder<R> {
     }
 
     fn color_type(&self) -> color::ColorType {
-        color::ColorType::L8
+        color::ColorType::Rgb8
     }
 
     fn into_reader(self) -> ImageResult<Self::Reader> {
-        Ok(WebpReader(Cursor::new(self.frame.ybuf), PhantomData))
+        Ok(WebpReader(Cursor::new(self.frame.to_rgb_vec()), PhantomData))
     }
 
     fn read_image(self, buf: &mut [u8]) -> ImageResult<()> {
         assert_eq!(u64::try_from(buf.len()), Ok(self.total_bytes()));
-        buf.copy_from_slice(&self.frame.ybuf);
+        buf.copy_from_slice(&self.frame.to_rgb_vec());
```

Comment:
Could we avoid this copy?

Reply:
How would we do that, could we return an iterator instead of a vector to convert from YUV to RGB?

Reply:
I think what he meant was that we are performing two copies: Once to construct the return value of `to_rgb_vec` which also takes an additional allocation, an then again when we copy that data to the output slice. Note the implementation of `to_rgb_vec` just requires access to a slice after it has been allocated. So instead there could be a helper function `fill_rgb(&self, _: &mut [u8])` which is called by both this and `to_rgb_vec`.